source/entities/ISelectable.hx

```diff
@@ -7,39 +7,46 @@ import flixel.util.FlxSpriteUtil.LineStyle;
 import flixel.group.FlxGroup;
 import flixel.util.FlxColor;
 import flixel.FlxSprite;
+import render.WorldConstants;
 
 using flixel.util.FlxSpriteUtil;
 
 class ISelectable extends FlxSpriteGroup {
-	private var _highlightCircle:FlxSprite;
+	private var _range:FlxSprite;
 	private var _isSelected:Bool;
 
-	public function new() {
+	public function new(radius:Float) {
 		super();
-		_highlightCircle = new FlxSprite();
-		var circHeight = 30;
-		var circWidth = 80;
-		_highlightCircle.makeGraphic(circWidth + 2, circHeight + 2, FlxColor.TRANSPARENT, true);
-		var lineStyle:LineStyle = {color: FlxColor.BLUE, thickness: 1};
+		_range = new FlxSprite();
+
+		var rangeRadii = 32;
+		var rangeWidth = rangeRadii * 2;
+		var rangeHeight = Std.int(rangeRadii * 2 * WorldConstants.HEIGHT_RATIO);
+		_range = new FlxSprite();
+		_range.makeGraphic(rangeWidth, rangeHeight, FlxColor.TRANSPARENT, true);
+		var lineStyle:LineStyle = {color: FlxColor.RED, thickness: 1};
 		var drawStyle:DrawStyle = {smoothing: true};
-		_highlightCircle.drawEllipse(0, 0, circWidth, circHeight, FlxColor.fromRGB(150, 150, 230, 128), lineStyle, drawStyle);
-		_highlightCircle.x = _highlightCircle.width / -2.0;
-		_highlightCircle.y = _highlightCircle.height / -2.0;
+		_range.drawEllipse(0, 0, rangeWidth, rangeHeight, FlxColor.fromRGB(255, 0, 0, 128), lineStyle, drawStyle);
+		_range.x = _range.width / -2;
+		
+		add(_range);
 
-		add(_highlightCircle);
 		_isSelected = false;
-		_highlightCircle.exists = _isSelected;
+		_range.visible = _isSelected;
```

Comment:
Toggling `visible` instead of `exists` so that the collisions work when it's not rendering